Google.PowerShell/BigQuery/BqJob.cs

```diff
@@ -415,12 +503,39 @@ public Apis.Bigquery.v2.Data.Job DoLoad()
 
         /// <summary>
         /// Extract Job main processing function.
+        /// *This is written using Apis.BigQuery.v2 becuase Cloud.BigQuery did not 
+        /// support Extract opertations at the time of writing.
         /// </summary>
         public Apis.Bigquery.v2.Data.Job DoExtract()
         {
             throw new NotImplementedException(
                 "Extract jobs are not implemented yet.  Use Get-BqTabledata instead.");
         }
+
+        /// <summary>
+        /// This function waits for a job to reach the "DONE" status and then returns.
+        /// </summary>
+        /// <param name="job">Job to poll for completion.</param>
+        public Apis.Bigquery.v2.Data.Job PollForCompletion(Apis.Bigquery.v2.Data.Job job)
```

Comment:
Instead of reinventing the wheel, maybe you want to just do Client.GetJob(job.JobReference).PollUntilCompleted().